core/src/main/java/hudson/cli/DisablePluginCommand.java

```diff
@@ -68,68 +78,161 @@ protected int run() throws Exception {
+        } catch (IllegalArgumentException iae) {
+            throw new IllegalArgumentException(hudson.cli.Messages.DisablePluginCommand_NoSuchStrategy(strategy, String.format("%s, %s, %s", PluginWrapper.PluginDisableStrategy.NONE, PluginWrapper.PluginDisableStrategy.MANDATORY, PluginWrapper.PluginDisableStrategy.ALL)));
+        }
+
+        // disable...
+        List<PluginWrapper.PluginDisableResult> results = jenkins.pluginManager.disablePlugins(strategyToUse, pluginNames);
+
+        // print results ...
+        printResults(results);
+
+        // restart if it was required and it's necessary (some plugin was disabled in this execution)
+        restartIfNecessary(results);
+
+        // return the appropriate error code
+        return getResultCode(results);
+    }
+
+    /**
+     * Print the result of all the process
+     * @param results the list of results for the disablement of each plugin
+     */
+    private void printResults( List<PluginWrapper.PluginDisableResult> results) {
+        for (PluginWrapper.PluginDisableResult oneResult : results) {
+            printResult(oneResult, 0);
+        }
+    }
+
+    /**
+     * Print indented the arguments with the format passed beginning with the indent passed.
+     * @param indent number of spaces at the beginning.
+     * @param format format as in {@link String#format(String, Object...)}
+     * @param arguments arguments to print as in {@link String#format(String, Object...)}
+     */
+    private void printIndented(int indent, String format, String... arguments) {
+        if (indent == 0) {
+            stdout.format(format + "%n", (Object[]) arguments);
+        } else {
+            String f = "%" + indent + "s" + format + "%n";
+            String[] newArgs = new String[arguments.length + 1];
+            newArgs[0] = " "; int i = 1;
+            for (String arg: arguments ) {
+                newArgs[i] = arg;
+                i++;
             }
+            stdout.format(f, newArgs);
         }
+    }
 
-        if (restart && somePluginDisabled) {
-            jenkins.safeRestart();
+    /**
+     * Print the result of a plugin disablement with the indent passed.
+     * @param oneResult the result of the disablement of a plugin.
+     * @param indent the initial indent.
+     */
+    private void printResult(PluginWrapper.PluginDisableResult oneResult, int indent) {
+        PluginWrapper.PluginDisableStatus status = oneResult.getStatus();
+        if(quiet && (PluginWrapper.PluginDisableStatus.DISABLED.equals(status) || PluginWrapper.PluginDisableStatus.ALREADY_DISABLED.equals(status))) {
+            return;
         }
 
-        return result;
+        printIndented(indent, Messages.DisablePluginCommand_StatusMessage(oneResult.getPlugin(), oneResult.getStatus(), oneResult.getMessage()));
+        if(oneResult.getDependantsDisableStatus().size() > 0) {
+            indent = (indent == 0) ? INDENT_SPACE : indent*2;
```

Comment:
nit: `indent * 2` (add spaces around operators)

Reply:
BTW, it was an error. It should be _indent + INDENT_SPACE_:
* indent * 2: 3, 6, 12, 24, 48, ...
* indent + INDENT_SPACE: 3, 6, 9, 12, 15, 18, ...

:-D